presto-orc/src/main/java/com/facebook/presto/orc/OrcSelectiveRecordReader.java

```diff
@@ -164,29 +170,95 @@ public OrcSelectiveRecordReader(
             this.constantValues[zeroBasedIndices.get(entry.getKey())] = entry.getValue();
         }
 
+        Map<Integer, Map<Subfield, TupleDomainFilter>> filtersWithNewIndices = filters
+                .entrySet()
+                .stream()
+                .collect(toImmutableMap(entry -> zeroBasedIndices.get(entry.getKey()), Map.Entry::getValue ))
+                ;
+
         // Initial order of stream readers is:
-        //  - readers with simple filters
+        //  - readers with integer equality
+        //  - readers with integer range / multivalues / inequality
+        //  - readers with filters
         //  - followed by readers for columns that provide input to filter functions
         //  - followed by readers for columns that doesn't have any filtering
-        streamReaderOrder = orderStreamReaders(columnTypes.keySet().stream().filter(index -> this.constantValues[index] == null).collect(toImmutableSet()), columnsWithFilters, filterFunctionInputs);
+        streamReaderOrder = orderStreamReaders(columnTypes.keySet().stream().filter(index -> this.constantValues[index] == null).collect(toImmutableSet()), filtersWithNewIndices, filterFunctionInputs);
     }
 
-    private static int[] orderStreamReaders(Collection<Integer> columnIndices, Set<Integer> columnsWithFilters, Set<Integer> filterFunctionInputs)
+    private static int scoreFilter(Map<Subfield, TupleDomainFilter> filters){
+        final int lowestScore = 10;
```

Comment:
- we don't use `final` for local variables
- constants need to be defined as `private static final` member variables and names should be all caps with underscore separators, e.g. `private static final int LOWEST_SCORE = 10;`
- in this case though, I think it is better to inline these values.

```
    private static int scoreFilter(Map<Subfield, TupleDomainFilter> filters)
    {
        checkArgument(!filters.isEmpty());

        if (filters.size() > 1){
            // Complex type column. Complex types are expensive!
            return 1000;
        }

        Map.Entry<Subfield, TupleDomainFilter> filterEntry = Iterables.getOnlyElement(filters.entrySet());
        if (!filterEntry.getKey().getPath().isEmpty()){
            // Complex type column. Complex types are expensive!
            return 1000;
        }

        TupleDomainFilter filter = filterEntry.getValue();
        if(filter instanceof BigintRange) {
            if(((BigintRange) filter).isSingleValue()){
                // Integer equality. Generally cheap.
                return 10;
            }
            return 50;
        }

        if(filter instanceof BigintValues || filter instanceof BigintMultiRange) {
            return 50;
        }

        return 100;
    }
```

Reply:
great! Good points